scipy/ndimage/tests/test_interpolation.py

```diff
@@ -538,418 +654,459 @@ def test_map_coordinates_endianness_with_output_parameter(self):
         # affine transform given output ndarray or dtype with either endianness
         # see issue #4127
-        data = np.array([1])
+        # NB: byteorder is numpy-specific
+        data = np.asarray([1])
         for out in [data.dtype, data.dtype.newbyteorder(),
                     np.empty_like(data),
                     np.empty_like(data).astype(data.dtype.newbyteorder())]:
-            returned = ndimage.affine_transform(data, [[1]], output=out)
+            returned = ndimage.affine_transform(data, np.asarray([[1]]), output=out)
             result = out if returned is None else returned
-            assert_array_almost_equal(result, [1])
+            assert_array_almost_equal(result, np.asarray([1]))
 
-    def test_affine_transform_output_shape(self):
+    def test_affine_transform_output_shape(self, xp):
         # don't require output_shape when out of a different size is given
-        data = np.arange(8, dtype=np.float64)
-        out = np.ones((16,))
+        if not is_numpy(xp):
+            pytest.xfail("`out` of a different size in NumPy-specific.")
 
-        ndimage.affine_transform(data, [[1]], output=out)
+        data = xp.arange(8, dtype=xp.float64)
+        out = xp.ones((16,))
+
+        ndimage.affine_transform(data, xp.asarray([[1]]), output=out)
         assert_array_almost_equal(out[:8], data)
 
         # mismatched output shape raises an error
         with pytest.raises(RuntimeError):
             ndimage.affine_transform(
                 data, [[1]], output=out, output_shape=(12,))
 
-    def test_affine_transform_with_string_output(self):
-        data = np.array([1])
-        out = ndimage.affine_transform(data, [[1]], output='f')
-        assert_(out.dtype is np.dtype('f'))
-        assert_array_almost_equal(out, [1])
+    def test_affine_transform_with_string_output(self, xp):
+        if not is_numpy(xp):
+            pytest.xfail("string output is numpy-specific")
+
+        data = xp.asarray([1])
+        out = ndimage.affine_transform(data, xp.asarray([[1]]), output='f')
+        assert out.dtype is np.dtype('f')
+        assert_array_almost_equal(out, xp.asarray([1]))
 
     @pytest.mark.parametrize('shift',
                              [(1, 0), (0, 1), (-1, 1), (3, -5), (2, 7)])
     @pytest.mark.parametrize('order', range(0, 6))
-    def test_affine_transform_shift_via_grid_wrap(self, shift, order):
+    def test_affine_transform_shift_via_grid_wrap(self, shift, order, xp):
         # For mode 'grid-wrap', integer shifts should match np.roll
-        x = np.array([[0, 1],
+        x = xp.asarray([[0, 1],
                       [2, 3]])
-        affine = np.zeros((2, 3))
-        affine[:2, :2] = np.eye(2)
-        affine[:, 2] = shift
+        affine = xp.zeros((2, 3))
+        affine[:2, :2] = xp.eye(2)
+        affine[:, 2] = xp.asarray(shift)
+
+        expected = np.roll(np.asarray(x), shift, axis=(0, 1))
+        expected = xp.asarray(expected)
```

Comment:
https://github.com/scipy/scipy/pull/21091#discussion_r1676804014 is being deferred to this PR and should be fixed here - no need to `xp.asarray` then  `np.asarray` `x` before it is used.

Reply:
This has been fixed: https://github.com/scipy/scipy/pull/21150/files#diff-3a88694edba2c070f37a8ec5f066aa515ae86d39a2f971fe74fe97e4c1579d21L881